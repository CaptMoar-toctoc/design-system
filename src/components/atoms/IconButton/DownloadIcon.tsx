import React from "react";
interface DownloadButtonProps {
  fill?: string,
  height?: number,
  width?: number
}

function DownloadIcon({ fill = '#3200C1', height = 20, width = 20 }: DownloadButtonProps) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      fill="none"
      viewBox="0 0 20 20"
    >
      <path
        fill={fill}
        fillRule="evenodd"
        d="M6.277 8.152a.625.625 0 01.884 0L10 10.992l2.84-2.84a.625.625 0 01.883.884l-3.281 3.28a.625.625 0 01-.884 0l-3.281-3.28a.625.625 0 010-.884z"
        clipRule="evenodd"
      ></path>
      <path
        fill={fill}
        fillRule="evenodd"
        d="M10 2.5c.345 0 .625.28.625.625v8.75a.625.625 0 11-1.25 0v-8.75c0-.345.28-.625.625-.625z"
        clipRule="evenodd"
      ></path>
      <path
        fill={fill}
        fillRule="evenodd"
        d="M3.125 11.25c.345 0 .625.28.625.625v4.375h12.5v-4.375a.625.625 0 111.25 0v4.375a1.25 1.25 0 01-1.25 1.25H3.75a1.25 1.25 0 01-1.25-1.25v-4.375c0-.345.28-.625.625-.625z"
        clipRule="evenodd"
      ></path>
    </svg>
  );
}

export default DownloadIcon;