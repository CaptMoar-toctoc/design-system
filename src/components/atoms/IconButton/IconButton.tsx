import React, { FC } from 'react';
import { iProps } from './iProps';
import { IconButton as IconButtonMUI, ThemeProvider } from '@mui/material';
import theme from '../../theme/theme';

const IconButton: FC<iProps> = ({
	children,
	handlerFn,
	iconAriaLabel,
	size
}) => {
	return (
		<ThemeProvider theme={theme}>
			<IconButtonMUI
				onClick={handlerFn}
				aria-label={iconAriaLabel}
				size={size}
			>
				{children}
			</IconButtonMUI>
		</ThemeProvider>
	);
}
export default IconButton;