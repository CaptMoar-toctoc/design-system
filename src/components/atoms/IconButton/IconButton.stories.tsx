import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import IconButton from './IconButton';
import DownloadIcon from './DownloadIcon';

export default {
    title: 'TocToc-DS/Atoms',
    component: IconButton,
    argTypes: {
        size: {
            control: {
                type: 'inline-radio',
                options: ['small', 'medium'],
            },
        },
    }
} as ComponentMeta<typeof IconButton>;

const Template: ComponentStory<typeof IconButton> = (args) => <IconButton {...args} />;

export const IconButtonTT = Template.bind({});

IconButtonTT.args = {
    children: <DownloadIcon />,
    iconAriaLabel: 'download',
    href: '',
    size: 'medium',
};