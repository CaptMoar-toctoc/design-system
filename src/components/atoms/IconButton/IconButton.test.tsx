import IconButton from './IconButton';
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import DownloadIcon from './DownloadIcon';

describe("IconButton testing", () => {
    const initProps = {
        children: <DownloadIcon />,
        handlerFn: () => { },
        iconAriaLabel: 'icon-button'
    };

    test('Render IconButton component', () => {
        render(<IconButton {...initProps} />)
    })

    test("Render SnapShot", () => {
        expect(render(<IconButton {...initProps} />)).toMatchSnapshot();
    })

    test.todo("Is a clickable component"), () => {
        render(<IconButton {...initProps} />)
        const iconButton = screen.getByRole('button')
        fireEvent.click(iconButton)
        expect(iconButton).toBeTruthy();
    }
})

export { }