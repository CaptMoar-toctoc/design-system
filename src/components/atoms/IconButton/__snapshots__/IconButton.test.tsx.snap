// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton testing Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="icon-button"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.277 8.152a.625.625 0 01.884 0L10 10.992l2.84-2.84a.625.625 0 01.883.884l-3.281 3.28a.625.625 0 01-.884 0l-3.281-3.28a.625.625 0 010-.884z"
            fill="#3200C1"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M10 2.5c.345 0 .625.28.625.625v8.75a.625.625 0 11-1.25 0v-8.75c0-.345.28-.625.625-.625z"
            fill="#3200C1"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M3.125 11.25c.345 0 .625.28.625.625v4.375h12.5v-4.375a.625.625 0 111.25 0v4.375a1.25 1.25 0 01-1.25 1.25H3.75a1.25 1.25 0 01-1.25-1.25v-4.375c0-.345.28-.625.625-.625z"
            fill="#3200C1"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="icon-button"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.277 8.152a.625.625 0 01.884 0L10 10.992l2.84-2.84a.625.625 0 01.883.884l-3.281 3.28a.625.625 0 01-.884 0l-3.281-3.28a.625.625 0 010-.884z"
          fill="#3200C1"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M10 2.5c.345 0 .625.28.625.625v8.75a.625.625 0 11-1.25 0v-8.75c0-.345.28-.625.625-.625z"
          fill="#3200C1"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M3.125 11.25c.345 0 .625.28.625.625v4.375h12.5v-4.375a.625.625 0 111.25 0v4.375a1.25 1.25 0 01-1.25 1.25H3.75a1.25 1.25 0 01-1.25-1.25v-4.375c0-.345.28-.625.625-.625z"
          fill="#3200C1"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
